src/Session.php

```diff
@@ -861,17 +861,19 @@ public static function getPreferredLanguage(): string
         /** @var array $CFG_GLPI */
         global $CFG_GLPI;
 
-       // Extract accepted languages from headers
-       // Accept-Language: fr-FR, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5
+        // Extract accepted languages from headers
+        // Accept-Language: fr-FR, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5
         $accepted_languages = [];
-        $values = explode(',', $_SERVER['HTTP_ACCEPT_LANGUAGE'] ?? '');
-        foreach ($values as $value) {
-            $parts = explode(';q=', trim($value));
-            $language = str_replace('-', '_', $parts[0]);
-            $qfactor  = $parts[1] ?? 1; //q-factor defaults to 1
-            $accepted_languages[$language] = $qfactor;
-        }
-        arsort($accepted_languages); // sort by qfactor
+        if (GLPI_ENVIRONMENT_TYPE !== 'testing') {
```

Comment:
You can open this if block at line 863 and close it at line 883. There is no need to execute the `foreach (array_keys($accepted_languages) as $language) {` loop if you prevent anything to be added to the `$accepted_languages` array.